src/audio/module_adapter/CMakeLists.txt

```diff
@@ -2,9 +2,9 @@
 
 
 if(CONFIG_IPC_MAJOR_3)
-	add_local_sources(sof module_adapter.c module_adapter_ipc3.c module/generic.c)
+	add_local_sources(sof module_adapter.c module_adapter_ipc3.c module/generic_ipc3.c)
 elseif(CONFIG_IPC_MAJOR_4)
-	add_local_sources(sof module_adapter.c module_adapter_ipc4.c module/generic.c)
+	add_local_sources(sof module_adapter.c module_adapter_ipc4.c module/generic_ipc4.c)
```

Comment:
after this commit generic_ipc3.c and generic_ipc4.c are 90% identical. I'm not a fan of such duplication.

Reply:
generic.c is a little special, it does not like other common files(extract small function to cover).
it difference are quite a lot(>10), and not easy extract small functions to cover.
Do you have any better ideas?

Reply:
you might need to handle each of them individually. In fact I'm not sure I understand the approach: is it our rule now that we must not have a single `#if CONFIG_IPC_MAJOR_3` in our C files, and if we find one, that file should be split? We absolutely must get rid of all of them? Why are they so harmful?

Reply:
there are two reasons to do this:

1. separate ipc3 and ipc4 code.
2. easier to optimize ipc4 specific code, no need have concern on impact on ipc3.

Reply:
@btian1 we still need to have a common file where any code that is teh same for ipc3/4 resides.